make/Main.gmk

```diff
@@ -970,7 +970,14 @@ else
   jdk.jdeps-gendata: java
 
   # The ct.sym generation uses all the moduleinfos as input
-  jdk.compiler-gendata: $(GENSRC_MODULEINFO_TARGETS)
+  jdk.compiler-gendata: $(GENSRC_MODULEINFO_TARGETS) $(JAVA_TARGETS)
+  ifeq ($(CREATE_BUILDJDK), true)
```

Comment:
I think I've sorted this out. The `buildjdk` COMPILER resolves to the newly built compiler if we do not have CREATE_BUILDJDK (which really means "USE_BUILDJDK"), and since they are built by jdk.compiler-launchers, we need to make that target first. Otoh, if we have a buildjdk, we're fine, unless we're in the process of creating a buildjdk, then we need to make sure we got the buildjdk first. Is this a proper reading of this?
(I think this part could do with some comments, since the logic here is not exactly crystal clear...)

But why would we even create ct-sym when creating a buildjdk? That seems like just a waste of time?

Reply:
If cross compiling, a BUILD_JDK can either be supplied by configure, or be built dynamically for the build platform. When not cross compiling, it's created implicitly by the build. There are two variables from configure that describe this:

`CREATE_BUILDJDK` : true if we are cross compiling and need to explicitly create a build jdk for the build platform.
`EXTERNAL_BUILDJDK`: true if a prebuilt build jdk has been supplied to configure.

When CREATE_BUILDJDK is true, there is also the variable `CREATING_BUILDJDK` that is set men `Main.gmk` is called recursively to actually create a build jdk for the build platform.

There are more comments for these cases around the jmod targets. However, while writing this I realize that we are missing a conditional on `EXTERNAL_BUILDJDK` being false.